lib/validators/datetime_setting_validator.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class DatetimeSettingValidator
+  def initialize(opts = {})
+    @opts = opts
+  end
+
+  def valid_value?(val)
+    return true if val.blank?
+
+    begin
+      # Only accept ISO 8601 datetime strings with timezone
+      # Must contain 'T' (date/time separator) and timezone indicator (Z or offset)
+      has_time_separator = val.include?("T")
+      has_timezone = val.end_with?("Z") || val.match?(/[+-]\d{2}:\d{2}$/)
+      return false unless has_time_separator && has_timezone
```

Comment:
Would parsing `val` with the expected format be easier?

Reply:
I guess you're trying to avoid the "expensive" exception that might be generated by `DateTime.iso8601`? Not sure it's worth it TBH